plugin/src/main/java/com/google/errorprone/xplat/checker/XplatBans.java

```diff
@@ -0,0 +1,359 @@
+import com.google.errorprone.bugpatterns.BugChecker.VariableTreeMatcher;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.source.tree.MethodTree;
+import com.sun.source.tree.NewClassTree;
+import com.sun.source.tree.Tree;
+import com.sun.source.tree.VariableTree;
+import com.sun.tools.javac.code.Symbol;
+import com.sun.tools.javac.code.Symbol.MethodSymbol;
+import com.sun.tools.javac.code.Symbol.VarSymbol;
+import com.sun.tools.javac.code.Type;
+import com.google.common.base.Charsets;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Optional;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+
+/**
+ * Check for usage of some Joda-Time classes and packages, which can be found in
+ * resources/Xplatbans.json. These calls are banned due to their incompatibility with cross platform
+ * development. Additional classes can be banned using the command line argument {@code
+ * -XepOpt:XplatBans:JSON=X}, where X is the path to a JSON file containing custom bans. Errors can
+ * be suppressed with the {@link XplatBanSuppression} annotation.
+ */
+@BugPattern(
+    name = "XplatBans",
+    summary = "Bans the usage of certain Joda-Time classes and packages for cross platform use.",
+    explanation =
+        "The usage of several Joda-Time classes and packages are banned from cross"
+            + " platform development due to incompatibilities. They are unsupported on the web"
+            + " and should also not be used on supported platforms. Additional bans can be configured"
+            + " with the command line argument -XepOpt:XplatBans:JSON=X, where X is the path to"
+            + " a JSON file containing custom bans.",
+    severity = ERROR,
+    suppressionAnnotations = XplatBanSuppression.class)
+public class XplatBans extends BugChecker
+    implements MethodInvocationTreeMatcher, NewClassTreeMatcher, VariableTreeMatcher,
+    MethodTreeMatcher {
+
+  private final Map<String, String> packageNames = new HashMap<>();
+
+  private final Map<String, String> classNames = new HashMap<>();
+
+  private final Map<String, Map<String, String>> methodNames = new HashMap<>();
+
+
+  /**
+   * A helper function that iterates though the JSON keys and puts them inside a map.
+   *
+   * @param json The JSONObject to be iterated through.
+   * @param map  The map to be added to.
+   * @throws JSONException This function should never throw a JSONException, as the only time
+   *                       getString() is used, it is being used with keys returned from keys().
+   */
+  private void jsonToMap(JSONObject json, Map<String, String> map) throws JSONException {
```

Comment:
Ideally function names should allows start with verbs to help readers understand what this function does.

conside: addJsonToMap

Reply:
Fixed